### PR TITLE
[DMS-629] Allow leading and trailing spaces in Descriptor's CodeValue and Short Description

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: src/dms/core/EdFi.DataManagementService.Core/Validation/DocumentValidator.cs [hunk 1]
```diff
@@ -146,6 +157,52 @@ PruneResult PruneNullData(JsonNode? documentBody, EvaluationResults evaluationRe
...
         }
 
+        PruneResult PruneCodeValueWhitespaceData(JsonNode? documentBody, EvaluationResults evaluationResults)
+        {
+            if (documentBody == null)
+            {
+                return new PruneResult.NotPruned();
+            }
+
+            var allowedProperties = new HashSet<string> { "codeValue", "shortDescription" };
```
Comment (bradbanister) [inline src/dms/core/EdFi.DataManagementService.Core/Validation/DocumentValidator.cs:167]: Let's move this to where it's only initialized once.

File: src/dms/tests/EdFi.DataManagementService.Tests.E2E/Features/Descriptors/UpdateDescriptorsValidation.feature [hunk 2]
```diff
@@ -215,14 +209,8 @@ Feature: Update a Descriptor
...
                   {
                     "validationErrors": {
-                      "$.codeValue": [
-                        "codeValue cannot contain leading or trailing spaces."
-                      ],
                       "$.namespace": [
                         "namespace cannot contain leading or trailing spaces."
-                      ],
-                      "$.shortDescription": [
-                        "shortDescription cannot contain leading or trailing spaces."
```
Comment (bradbanister) [inline src/dms/tests/EdFi.DataManagementService.Tests.E2E/Features/Descriptors/UpdateDescriptorsValidation.feature:225]: This is good, but let's also add a test like your new create scenario 18, that updates successfully with whitespace and the whitespace is removed.

File: src/dms/core/EdFi.DataManagementService.Core/Validation/DocumentValidator.cs [hunk 3]
```diff
@@ -27,6 +27,8 @@ internal interface IDocumentValidator
 
 internal class DocumentValidator() : IDocumentValidator
 {
+    private readonly HashSet<string> allowedProperties = new HashSet<string> { "codeValue", "shortDescription" };
```
Comment (bradbanister) [inline src/dms/core/EdFi.DataManagementService.Core/Validation/DocumentValidator.cs:30]: Let's name this `trimmableProperties`. They're not just allowed properties.

File: src/dms/core/EdFi.DataManagementService.Core/Validation/DocumentValidator.cs [hunk 4]
```diff
@@ -64,6 +67,17 @@ private static JsonSchema GetSchema(ResourceSchema resourceSchema, RequestMethod
             results = resourceSchemaValidator.Evaluate(context.ParsedBody, validatorEvaluationOptions);
         }
 
+        var whiteSpacePruneResult = PruneCodeValueWhitespaceData(context.ParsedBody, results);
```
Comment (bradbanister) [inline src/dms/core/EdFi.DataManagementService.Core/Validation/DocumentValidator.cs:70]: whitespace is one word - `whitespacePruneResult`

**Review by bradbanister — CHANGES_REQUESTED** [review]
Two small final changes

**Review by bradbanister — APPROVED** [review]
